### DISCUSSION TITLE
Spacy dependency requirements are not met for pydantic versions ^2

### DISCUSSION THREAD

**Discussion by jacobmanderson** [explosion/spaCy#13198] [Help: Installation]
## How to reproduce the behaviour

Example pyproject.toml

```
[tool.poetry.dependencies]
python = "^3.7"
pydantic = "^2.0.0"
spacy = "^3.2.0"
```

When running `poetry update --lock` there is a build dependency error between pydantic version ^2 and typing-extensions.
```
And because spacy (3.7.2) depends on typing-extensions (>=3.7.4.1,<4.5.0)
 and pydantic (2.0) depends on typing-extensions (>=4.6.1), pydantic (2.0) is incompatible with spacy (>=3.2.0,<4.0.0).
``` 

The restriction on typing-extensions is in the `requirements.txt`, which are run via `setuptools` in `pyproject.toml`
## Your Environment

* Operating System: ubuntu 22.04
* Python Version Used: 3.10
* spaCy Version Used:3.7.2
* Environment Information:

**Comment by adrianeboyd** [comment]
It's difficult to support+test every possible combination of requirements and we've had trouble with older versions of `pydantic` and newer versions of `typing_extensions` in the past.

In this case I'd strongly suggest using python 3.8+, where `typing_extensions` is no longer a requirement for `spacy`. It's only needed for python 3.7 and earlier for backports.
